File: src/main/java/scrabble/phrases/words/Verb.java
package scrabble.phrases.words;

import java.util.Objects;

/**
 * Romanian verb.
 *
 * @param word the verb
 * @param syllables the number of syllables
 * @param rhyme the rhyme suffix (last 3 characters)
 */
public record Verb(
    String word,
    int syllables,
    String rhyme
) implements Word {

    /**
     * Creates a new Verb with computed syllables and rhyme.
     *
     * @param word the verb
     */
    public Verb(String word) {
        this(
            word,
            WordUtils.computeSyllableNumber(word),
            WordUtils.computeRhyme(word)
        );
    }

    /**
     * Compact constructor for validation.
     */
    public Verb {
        Objects.requireNonNull(word, "word cannot be null");
    }

    // Legacy method names for compatibility
    public String getWord() {
        return word;
    }

    public int getSyllables() {
        return syllables;
    }

    public String getRhyme() {
        return rhyme;
    }
}

File: src/main/kotlin/scrabble/phrases/words/Verb.kt
package scrabble.phrases.words

data class Verb(
    override val word: String,
    override val syllables: Int = WordUtils.computeSyllableNumber(word),
    override val rhyme: String = WordUtils.computeRhyme(word)
) : Word {
    init {
        require(word.isNotBlank()) { "verb word must not be blank" }
        require(word.length >= 2) { "verb word must have at least 2 characters" }
        require(word.any { it.isLetter() }) { "verb word must contain at least one letter" }
    }
}
